File: src/main/java/ua/notion/domain/entity/PlantInstance.java
package ua.notion.domain.entity;

import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PlantInstance {
  private Integer id;
  private Integer gardenId;
  private Integer plantId;
  private Integer cellX;
  private Integer cellY;
  private LocalDate plantedAt;
  private Integer growthStage;
  private Boolean isWatered;
  private Boolean isFertilized;
}

File: src/server/src/models/plant_instance.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct PlantInstance {
    pub id: i32,
    pub garden_id: i32,
    pub plant_id: i32,
    pub cell_x: i32,
    pub cell_y: i32,
    pub planted_at: NaiveDateTime,
    pub growth_stage: i32,
    pub is_watered: bool,
    pub is_fertilized: bool,
}
